lib/discord_bot/discord_bot.ex

```diff
@@ -0,0 +1,226 @@
+defmodule DiscordBot do
+  use Supervisor
+  alias Alchemy.Client
+  alias Alchemy.Embed
+
+  defmodule BotState do
```

Comment:
This `GenServer` may be better replaced by just using `Cachex` which we are using for other in-memory caching. You can register a new cache (which will incidentally start a new genserver for it) and then just read/write to it via the cache APIs.

Reply:
https://github.com/whitfin/cachex

Reply:
Much better!